lib/Conversion/ExportVerilog/PrepareForEmission.cpp

```diff
@@ -615,8 +615,7 @@ bool EmittedExpressionStateManager::shouldSpillWireBasedOnState(Operation &op) {
   // If the operation is only used by an assignment, the op is already spilled
   // to a wire.
   if (op.hasOneUse() &&
-      isa<hw::OutputOp, sv::AssignOp, sv::BPAssignOp, sv::PAssignOp>(
```

Comment:
Why was `sv::PAssignOp` removed here? Is this maybe an unrelated change that snuck in? I wouldn't think this is required for this change.

Reply:
This is unrelated to the PR but it's necessary for name propagation improvement. I'll split a PR.

Reply:
It doesn't have to be a separate PR if it's related, I'm just curious why it is necessary here, since we are only updating the canonicalizer or wires.

Reply:
The condition was intended to avoid spilling same expressions twice by skipping expressions that have single use of assignments. Therefore an expression used by procedural assignments was not spilled to a wire. For example, 
```
%0 = comb.add {sv.namehint = "good_name"}
sv.always .. {
   sv.passign %reg, %0 : ..
}
```
In this case `%0` was inlined to the assignment, but we want to create a wire for it.

Reply:
I see, so I think this makes sense then. In general, I guess it's all about tradeoffs, so trading off some inlining for better names sounds good to me.